### PR TITLE
#58 First work on benchmarking strategy

### PR DESCRIPTION
**Pull request by Roarster**

### CODE REVIEW

File: docs/benchmarking.md
Comment (adamconnelly) [inline docs/benchmarking.md:43]: I wonder if we can also compare it against the last successful build results so that if performance degrades by more than X% we can fail a check.

**Review by adamconnelly — APPROVED** [review]
Sounds great Ian - cheers.

I think the check will just be failing because of the commit message format. There's an explanation of it in the contrib guide - I'm hoping we can automatically generate release notes from the commit messages, so that's why I'm trying to use conventional commits. 

On that note, maybe we should add a link to this from the contribution guide?

**Comment by adamconnelly** [comment]
@Roarster just read this blog post about gRPC performance improvements in .NET 5: https://devblogs.microsoft.com/aspnet/grpc-performance-improvements-in-net-5/

It's maybe not directly relevant, but it's interesting to read the sorts of things they've done to improve performance, so might be worth adding a note somewhere so we can use it for inspiration.

**Comment by Roarster** [comment]
@adamconnelly I've added a bit more information on this. If everything is ok I'll fix the commits.

**Comment by adamconnelly** [comment]
That looks great. Give me a shout once the commits are sorted and I'll merge it. I'll also add you as a contributor so you can merge yourself - just haven't gotten round to doing that yet.

**Comment by adamconnelly** [comment]
@Roarster I dunno if this'll be more or less frustrating for you :smile_cat: but if you haven't already you can `npm install` in the repo and it'll setup a Husky check for the commit messages. That's me added you as a contributor, so feel free to just merge once it's all sorted.

**Comment by adamconnelly** [comment]
@Roarster sorry - I'd been assuming that the build was failing because the commit message wasn't right, but I hadn't actually looked at the build results. I had a look and realised it's now failing because the script doesn't work quite right when the PR branch is from a cloned repo rather than in this repo.

I've put together some changes that I think should sort it here: https://github.com/adamconnelly/Thrift.Net/pull/73

Any chance you can just pull / cherry-pick that commit into your branch and see if it sorts the check?

**Comment by Roarster** [comment]
@adamconnelly Yeah, that's fixed it. Do you want me to hold off until you've merged that separately and rebase before merging?

**Comment by adamconnelly** [comment]
@Roarster nah - it's the same result either way. Just merge this and I'll close that other PR. I just pushed it up so I could check the pipeline worked before messaging you.

**Pull request merged**
